### PR TITLE
rpc: add version to get_info

### PR DESCRIPTION
**Pull request by jtgrassie**
Expose the release version in get_info RPC response.

### CODE REVIEW

**Comment by moneromooo-monero** [comment]
Any opinions on this ? I originally did not include this in RPC because of fingerprinting and it's not really useful for RPC.

**Comment by selsta** [comment]
I think the use case for this was to only add v0.13.0.4 remote nodes to moneroworld.com because of the caching bug.

**Comment by jtgrassie** [comment]
It's useful for users of public remote nodes to ensure the node is running the latest point release. 

For context: https://www.reddit.com/r/Monero/comments/9yryab/18089_operators_please_always_run_the_most_recent/

**Comment by moneromooo-monero** [comment]
Why would you use a human readable string for this, rather than the RPC version, which is what is relevant for, er, RPC usage ?

**Comment by moneromooo-monero** [comment]
Oh, the bug didn't change RPC so no version change. OK.

**Comment by jtgrassie** [comment]
@moneromooo-monero I'm not buying the fingerprinting argument for a couple of reasons: 1) it's just the daemon version which everyone should be running anyway, 2) get_info has other more interesting fingerprinting data.

The reason to use MONERO_VERSION is it's a) parseable by `tools::vercmp` and b) it is not the RPC version, but the daemon version.

**Comment by xiphon** [comment]
> 1) it's just the daemon version which everyone should be running anyway

Nope. Any version that follows the current network consensus is fine.

And yep, this could be used for fingerprinting.

**Comment by Gingeropolous** [comment]
What kind of fingerprinting are we talking about? The provider of the public remote node is already exposing a lot ..

```
}user@user-VirtualBox:~/moneriote$ curl -X POST http://ushouldrunyourownnode.moneroworld.com:18089/getinfo -H 'Content-Type: application/json'
{
  "alt_blocks_count": 132,
  "block_size_limit": 600000,
  "block_size_median": 11162,
  "block_weight_limit": 600000,
  "block_weight_median": 11162,
  "bootstrap_daemon_address": "",
  "cumulative_difficulty": 22511576375988476,
  "database_size": 70226743296,
  "difficulty": 55999239125,
  "free_space": 18446744073709551615,
  "grey_peerlist_size": 4880,
  "height": 1709430,
  "height_without_bootstrap": 1709430,
  "incoming_connections_count": 37,
  "mainnet": true,
  "nettype": "mainnet",
  "offline": false,
  "outgoing_connections_count": 8,
  "rpc_connections_count": 41,
  "stagenet": false,
  "start_time": 0,
  "status": "OK",
  "target": 120,
  "target_height": 1709207,
  "testnet": false,
  "top_block_hash": "f333f71fd0539240fdc85260443ca176cf5a7f66f327349ed44e35e03f5afbb9",
  "tx_count": 3640869,
  "tx_pool_size": 16,
  "untrusted": false,
  "update_available": false,
  "was_bootstrap_ever_used": false,
  "white_peerlist_size": 1000

```

**Comment by moneromooo-monero** [comment]
And more in https://github.com/monero-project/monero/pull/4879, some of which I thought already were restricted...

**Comment by Gingeropolous** [comment]
I just checked that node, and i'm indeed running it with 

```rpc-restricted-bind-port=18089
```

so, either restricted RPC bind port isn't work right or not many things are as restricted as thought.

**Comment by m2049r** [comment]
can i tell from this information that the node is on the correct chain? (after an upgrade)

**Comment by jtgrassie** [comment]
@m2049r Not quite. This exposes 0.13.0.4-release. The hard_fork_info exposes the major block version. 

@moneromooo-monero I actually am coming round to the idea this should be restricted like much of the things you have in #4879. I'll push an update.

File: src/rpc/core_rpc_server_commands_defs.h [hunk 1]
```diff
@@ -926,6 +927,7 @@ namespace cryptonote
         KV_SERIALIZE(was_bootstrap_ever_used)
         KV_SERIALIZE(database_size)
         KV_SERIALIZE(update_available)
+        KV_SERIALIZE(version)
```
[the remark this answers is not in the document]
Comment (jtgrassie) [reply src/rpc/core_rpc_server_commands_defs.h]: Done

**Comment by Gingeropolous** [comment]
well now im confused. Will someone running --restricted-rpc-port X be able to share the daemon version now?

**Comment by iDunk5400** [comment]
Time to ask for `--unrestricted-methods=start_mining,stop_mining,...` and
`--unrestricted-info=version,rpc_connections_count,...` :)

**Comment by jtgrassie** [comment]
@Gingeropolous No, this is hidden now when running restricted (as are other fields per #4879). 

@iDunk5400 Probably not on this PR comment thread, but agreed it would be nice to have finer grained control over what to (un)restrict.

**Comment by moneromooo-monero** [comment]
Control over this is also a fingerprinting source :)

**Pull request merged**